pydy/tests/test_system.py

```diff
@@ -395,3 +397,27 @@ def test_integrate(self):
         sys = System(self.kane, times=times)
         with testing.assert_raises(NotImplementedError):
             sys.generate_ode_function(generator='made-up')
+
+
+class TestJointSystem():
+
+    def setup(self):
+        self.system = System.using_joints()
```

Comment:
It is just an odd pattern that typically Python users will be unfamiliar with. I like the other three options better than the first in that stackoverflow answer. I do not think that the other more typically solutions will introduce an unnecessary amount of bugs. If we want one system class, no sublcasses, then simply make all arguments to System be kwargs instead of args. If you get `method=None` then you instantiate it without a method. The user can then build the system and once ready, it will be able to generate the needed Method class behind the scenes.